### PR TITLE
sd-bus: sd_bus_call/sd_bus_call_async_docs + cleanups

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: man/sd_bus_call_method.xml [hunk 1]
```diff
@@ -59,15 +60,15 @@
...
+    bus message object and calling the corresponding D-Bus method. It combines the
     <citerefentry><refentrytitle>sd_bus_message_new_method_call</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
     <citerefentry><refentrytitle>sd_bus_message_append</refentrytitle><manvolnum>3</manvolnum></citerefentry> and
     <citerefentry><refentrytitle>sd_bus_call</refentrytitle><manvolnum>3</manvolnum></citerefentry>
     functions into a single function call.</para>
 
-    <para><function>sd_bus_call_method_async()</function> is a convenience function for calling a
-    D-Bus method asynchronously. It combines the
+    <para><function>sd_bus_call_method_async()</function> is a convenience function for initializing
+    a bus message calling and calling the corresponding D-Bus method asynchronously. It combines the
```
Comment (anitazha) [inline man/sd_bus_call_method.xml]: `a bus message calling` -> `a bus message object`?

File: man/sd_bus_call.xml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    runtime by changing the value of <varname>$SYSTEMD_BUS_TIMEOUT</varname>. Instead, call
+    <function> sd_bus_set_method_call_timeout</function> to change the default method call timeout.
+    </para>
+
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, these functions returns a non-negative integer. On failure, they return a
```
Comment (anitazha) [inline man/sd_bus_call.xml]: `these functions returns` -> `these functions return`

File: man/sd_bus_call.xml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    If <parameter>slot</parameter> is <constant>NULL</constant>, the lifetime of the method call is
+    bound to the lifetime of the bus object itself, and it can cannot not be cancelled
+    independently. See
+    <citerefentry><refentrytitle>sd_bus_slot_set_floating</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    for details. The <parameter>callback</parameter> function is called when the response arrives and
+    receives the response, <parameter>userdata</parameter> and a
+    <structname>sd_bus_error</structname> object as its arguments. The
+    <structname>sd_bus_error</structname> is unused here and should be ignored. If
+    <parameter>callback</parameter> returns a non-negative integer when called, a debug message is
+    logged along with details about the response.
```
Comment (anitazha) [inline man/sd_bus_call.xml]: This section is missing opening and closing `<para>`

File: man/sd_bus_call.xml [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+    <para>If <parameter>usec</parameter> is zero, the timeout set using
+    <citerefentry><refentrytitle>sd_bus_set_method_call_timeout</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    is used. If no method call timeout was set, the timeout is read from the
+    <varname>$SYSTEMD_BUS_TIMEOUT</varname> environment variable. If this environment variable is
+    unset or does not contain a valid timeout, the implementation falls back to a predefined method
+    call timeout of 25 seconds. Note that <varname>$SYSTEMD_BUS_TIMEOUT</varname> is read once and
+    cached so callers should not rely on being able to change the default method call timeout at
+    runtime by changing the value of <varname>$SYSTEMD_BUS_TIMEOUT</varname>. Instead, call
+    <function> sd_bus_set_method_call_timeout</function> to change the default method call timeout.
```
Comment (anitazha) [inline man/sd_bus_call.xml]: `<function> sd_bus_set_method_call_timeout</function>` -> `<function>sd_bus_set_method_call_timeout</function>` (extra whitespace)

File: man/sd_bus_call.xml [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_bus_call()</function> takes a complete bus message object and calls the
+    corresponding D-Bus method. The response is stored in <parameter>reply</parameter>.
+    <parameter>usec </parameter> indicates the timeout in microseconds. If
```
Comment (yuwata) [inline man/sd_bus_call.xml]: unnecessary white space after `usec`.

File: man/sd_bus_call.xml [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    <para><function>sd_bus_call_async()</function> is like <function>sd_bus_call()</function> but
+    works asynchronously. The <parameter>callback</parameter> shall reference a function to call
+    when the event source is triggered. The <parameter>userdata</parameter> pointer will be passed
+    to the callback function, and may be chosen freely by the caller. If <parameter>slot</parameter>
+    is not <constant>NULL</constant> and <function>sd_bus_call_async()</function> succeeds,
+    <parameter>slot</parameter> is set to a slot object which can be used to cancel the method call
+    at a later time using
+    <citerefentry><refentrytitle>sd_bus_slot_unref</refentrytitle><manvolnum>3</manvolnum></citerefentry>.
+    If <parameter>slot</parameter> is <constant>NULL</constant>, the lifetime of the method call is
+    bound to the lifetime of the bus object itself, and it can cannot not be cancelled
```
Comment (yuwata) [inline man/sd_bus_call.xml]: `can cannot not` -> `cannot`

**Comment by yuwata** [comment]
LGTM.

**Comment by keszybz** [comment]
Nice, thanks!

**Comment by keszybz** [comment]
I'll write up "sd_bus_open" and "sd_bus_open_with_description" now. It is one of the important man pages we are missing right now.

**Comment by keszybz** [comment]
Hmm, it's already described in `sd_bus_default`... The list at the top of this page is out-of-date. I'll try `sd_bus_start` instead ;)

**Pull request merged**